File: Android/remoteconfig/app/src/main/java/com/huawei/agc/quickstart/remoteconfig/RemoteConfigMessageService.java
/*
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.huawei.agc.quickstart.remoteconfig;

import android.content.SharedPreferences;
import android.util.Log;

import com.huawei.hms.push.HmsMessageService;
import com.huawei.hms.push.HmsMessaging;
import com.huawei.hms.push.RemoteMessage;

import java.util.Arrays;

public class RemoteConfigMessageService extends HmsMessageService {
    private static final String TAG = "MessageService";

    @Override
    public void onMessageReceived(RemoteMessage message) {
        Log.i(TAG, "onMessageReceived is called");
        if (message == null) {
            Log.e(TAG, "Received message entity is null!");
            return;
        }
        Log.i(TAG, "getCollapseKey: " + message.getCollapseKey()
                + "\n getData: " + message.getData()
                + "\n getFrom: " + message.getFrom()
                + "\n getTo: " + message.getTo()
                + "\n getMessageId: " + message.getMessageId()
                + "\n getOriginalUrgency: " + message.getOriginalUrgency()
                + "\n getUrgency: " + message.getUrgency()
                + "\n getSendTime: " + message.getSentTime()
                + "\n getMessageType: " + message.getMessageType()
                + "\n getTtl: " + message.getTtl());
        RemoteMessage.Notification notification = message.getNotification();
        if (notification != null) {
            Log.i(TAG, "\n getImageUrl: " + notification.getImageUrl()
                    + "\n getTitle: " + notification.getTitle()
                    + "\n getTitleLocalizationKey: " + notification.getTitleLocalizationKey()
                    + "\n getTitleLocalizationArgs: " + Arrays.toString(notification.getTitleLocalizationArgs())
                    + "\n getBody: " + notification.getBody()
                    + "\n getBodyLocalizationKey: " + notification.getBodyLocalizationKey()
                    + "\n getBodyLocalizationArgs: " + Arrays.toString(notification.getBodyLocalizationArgs())
                    + "\n getIcon: " + notification.getIcon()
                    + "\n getSound: " + notification.getSound()
                    + "\n getTag: " + notification.getTag()
                    + "\n getColor: " + notification.getColor()
                    + "\n getClickAction: " + notification.getClickAction()
                    + "\n getChannelId: " + notification.getChannelId()
                    + "\n getLink: " + notification.getLink()
                    + "\n getNotifyId: " + notification.getNotifyId());
        }

        if (message.getDataOfMap().containsKey("DATA_STATE")) {
            SharedPreferences preferences = this.getApplicationContext().getSharedPreferences("Remote_Config", MODE_PRIVATE);
            preferences.edit().putBoolean("DATA_OLD", true).apply();
        }
    }

    @Override
    public void onNewToken(String s) {
        super.onNewToken(s);
        Log.d(TAG, "Push Token : " + s);
        HmsMessaging.getInstance(this).subscribe("PUSH_REMOTE_CONFIG").addOnCompleteListener(task -> {
            if (task.isSuccessful()) {
                Log.i(TAG, "subscribe Complete");
            } else {
                Log.e(TAG, "subscribe failed: cause=" + task.getException().getMessage());
            }
        });
    }
}

File: Android/remoteconfig/kotlin-app/src/main/java/com/huawei/agconnect/kotlindemo/RemoteConfigMessageService.kt
/*
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.huawei.agconnect.kotlindemo

import android.util.Log
import com.huawei.hmf.tasks.Task
import com.huawei.hms.push.HmsMessageService
import com.huawei.hms.push.HmsMessaging
import com.huawei.hms.push.RemoteMessage

class RemoteConfigMessageService : HmsMessageService() {
    override fun onMessageReceived(message: RemoteMessage) {
        Log.i(TAG, "onMessageReceived is called")
        if (message == null) {
            Log.e(TAG, "Received message entity is null!")
            return
        }
        if (message.dataOfMap.containsKey("DATA_STATE")) {
            val preferences = this.applicationContext.getSharedPreferences("Remote_Config", MODE_PRIVATE)
            preferences.edit().putBoolean("DATA_OLD", true).apply()
        }
    }

    override fun onNewToken(s: String) {
        super.onNewToken(s)
        Log.d(TAG, "Push Token : $s")
        HmsMessaging.getInstance(this).subscribe("PUSH_REMOTE_CONFIG").addOnCompleteListener { task: Task<Void?> ->
            if (task.isSuccessful) {
                Log.i(TAG, "subscribe Complete")
            } else {
                Log.e(TAG, "subscribe failed: cause=" + task.exception.message)
            }
        }
    }

    companion object {
        private const val TAG = "MessageService"
    }
}